api/auth.go

```diff
@@ -461,6 +461,67 @@ func teamList(w http.ResponseWriter, r *http.Request, t auth.Token) error {
 	return json.NewEncoder(w).Encode(result)
 }
 
+// title: team info
+// path: /teams/{name}
+// method: GET
+// produce: application/json
+// responses:
+//   200: Info team
+//   404: Not found
+//   401: Unauthorized
+func teamInfo(w http.ResponseWriter, r *http.Request, t auth.Token) error {
+	teamName := r.URL.Query().Get(":name")
+	team, err := auth.GetTeam(teamName)
+	if err != nil {
+		return err
+	}
+	canRead := permission.Check(t, permission.PermTeamRead)
+	if !canRead {
+		return permission.ErrUnauthorized
+	}
+	apps, err := app.List(&app.Filter{
+		Extra:     map[string][]string{"teams": {team.Name}},
+		TeamOwner: team.Name})
+	if err != nil {
+		return err
+	}
+	pools, err := pool.ListPoolsForTeam(team.Name)
+	if err != nil {
+		return err
+	}
+	users, err := auth.ListUsers()
+	if err != nil {
+		return err
+	}
+	includedUsers := make([]*apiUser, 0)
+	for _, user := range users {
+		for _, roleInstance := range user.Roles {
+			role, err := permission.FindRole(roleInstance.Name)
```

Comment:
You could cache the result of this function, to avoid calling it multiple times for the same role

Reply:
I do not understand, Do I need sweep the user.Roles for get roleInstance.Name ?

Reply:
You are calling `permission.FindRole` for each role of each user. If a team has 10 different users with a common role, you would be making the same request 10 times.

A suggestion to improve this would be creating a map with the role names as keys, and the roles as values. Before calling `permission.FindRole`, you could check if that value is already in the map to avoid these unnecessary requests.

Here's an example of how you could do it:

```
cachedRoles := make(map[string]Role)
for _, user := range users {
	for _, roleInstance := range user.Roles {
		role, ok := cachedRoles[roleInstance.Name]
		if !ok {
			role, err = permission.FindRole(roleInstance.Name)
			if err != nil {...}
			cachedRoles[roleInstance.Name] = role
		}
		...
	}
}
```